iceoryx_posh/test/moduletests/test_popo_condition_variable.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/internal/popo/waitset/condition_variable_data.hpp"
+#include "iceoryx_posh/internal/popo/waitset/condition_variable_signaler.hpp"
+#include "iceoryx_posh/internal/popo/waitset/condition_variable_waiter.hpp"
+
+#include "test.hpp"
+#include <memory>
+#include <thread>
+#include <atomic>
+
+using namespace ::testing;
+using ::testing::Return;
+using namespace iox::popo;
+using namespace iox::cxx;
+using namespace iox::units::duration_literals;
+
+class ConditionVariable_test : public Test
+{
+  public:
+    ConditionVariableData m_condVarData;
+    ConditionVariableWaiter m_waiter{&m_condVarData};
+    ConditionVariableSignaler m_signaler{&m_condVarData};
+
+    iox::posix::Semaphore m_syncSemaphore = iox::posix::Semaphore::create(0u).get_value();
+
+    void SetUp(){};
+    void TearDown()
+    {
+        // Reset condition variable
+        m_waiter.reset();
+    };
+};
+
+TEST_F(ConditionVariable_test, TimedWaitWithInvalidTimeResultsInFailure)
+{
```

Comment:
How many waiters and signalers are allowed for one ConditionVariableData? Are those limits tested?

Reply:
I'm not aware of any additional limits compared with a `Semaphore`. If any, limits should be tested in the `Semaphore` unit test.